app/controllers/schedules_controller.rb

```diff
@@ -52,9 +53,17 @@ def events
     @events_schedules = @program.selected_event_schedules(
       includes: [:room, { event: %i[track event_type speakers submitter] }]
     )
+
+    @events_schedules = @events_schedules.select{ |e| e.event.favourite_users.exists?(current_user.id) } if @events_schedules && current_user && @favourites
```

Comment:
This may be a slightly different use case, but I think related enough...
How would you feel about showing a schedule which is favorite events + events where you are speaking? 

I am thinking about creating an `event_schedule_for_user(user)` method, which is essentially your personal conference agenda. My app has assigned volunteers which could also how up there.